test/extended/util/volumes.go

```diff
@@ -1,125 +1,29 @@
 package util
 
 import (
-	"errors"
-	"fmt"
 	"strings"
 
-	kapiv1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
 	e2e "k8s.io/kubernetes/test/e2e/framework"
 )
 
-// CreateNFSBackedPersistentVolume creates a persistent volume backed by a pod
-// running nfs
-func CreateNFSBackedPersistentVolume(name, namespace, capacity, server string, number int) *kapiv1.PersistentVolume {
-	e2e.Logf("Creating persistent volume %s", name)
-	return &kapiv1.PersistentVolume{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "PersistentVolume",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   name,
-			Labels: map[string]string{"name": name, "namespace": namespace},
-		},
-		Spec: kapiv1.PersistentVolumeSpec{
-			PersistentVolumeSource: kapiv1.PersistentVolumeSource{
-				NFS: &kapiv1.NFSVolumeSource{
-					Server:   server,
-					Path:     fmt.Sprintf("/exports/data-%d", number),
-					ReadOnly: false,
-				},
-			},
-			PersistentVolumeReclaimPolicy: kapiv1.PersistentVolumeReclaimDelete,
-			Capacity: kapiv1.ResourceList{
-				kapiv1.ResourceStorage: resource.MustParse(capacity),
-			},
-			AccessModes: []kapiv1.PersistentVolumeAccessMode{
-				kapiv1.ReadWriteMany,
-				kapiv1.ReadWriteOnce,
-				kapiv1.ReadOnlyMany,
-			},
-		},
-	}
-}
-
-// SetupNFSBackedPersistentVolumes sets up an NFS backed persistent volume
-// Currently, the image that is used exports 10 nfs shares, so the maximum
-// integer for count would be 10
-func SetupNFSBackedPersistentVolumes(oc *CLI, capacity string, count int) (volumes []*kapiv1.PersistentVolume, err error) {
-	e2e.Logf("Setting up nfs backed persistent volume(s)")
-
-	maxVolumes := 10
-	if count > maxVolumes {
-		return nil, errors.New(fmt.Sprintf("SetupNFSBackedPersistentVolumes only supports a maximum of %d volumes, you specified %d", maxVolumes, count))
-	}
-
-	c := oc.AdminKubeClient().Core().PersistentVolumes()
-	prefix := oc.Namespace()
-
-	_, svc, err := SetupNFSServer(oc, capacity)
+func DeletePVCsForDeployment(client clientset.Interface, oc *CLI, deploymentPrefix string) {
+	pvclist, err := client.CoreV1().PersistentVolumeClaims(oc.Namespace()).List(metav1.ListOptions{})
 	if err != nil {
-		return nil, err
+		e2e.Logf("pvc list error %#v\n", err)
 	}
-	server, err := oc.Run("get").Args("service", svc.Name, "--template", "{{.spec.clusterIP}}").Output()
-	if err != nil {
-		return nil, err
-	}
-
-	for i := 0; i < count; i++ {
-		pv, err := c.Create(CreateNFSBackedPersistentVolume(fmt.Sprintf("%s%s-%d", pvPrefix, prefix, i), oc.Namespace(), capacity, server, i))
-		if err != nil {
-			e2e.Logf("Error creating persistent volume %v\n", err)
-			return nil, err
+	for _, pvc := range pvclist.Items {
+		e2e.Logf("found pvc %s\n", pvc.Name)
+		if strings.HasPrefix(pvc.Name, "mongo") {
```

Comment:
"mongo" should be deploymentPrefix ?

Reply:
yes it should

Reply:
and i accidentally re-added it as well

Reply:
sorry, that was meant for the mysql_replica.go comment